system/lib/libc/emscripten_memcpy_bulkmem.S

```diff
@@ -7,10 +7,16 @@
 .globl _emscripten_memcpy_bulkmem
 _emscripten_memcpy_bulkmem:
   .functype _emscripten_memcpy_bulkmem (PTR, PTR, PTR) -> (PTR)
-  local.get 0
-  local.get 1
   local.get 2
-  memory.copy 0, 0
+#ifdef __wasm64__
+  i32.wrap_i64
+#endif
+  if
+    local.get 0
+    local.get 1
+    local.get 2
```

Comment:
Maybe a comment here?   "Unlike memcpy itself the memory copy instruction doesn't support OOB zero-length copies so we need an explicit check here".